lib/teleterm/clusters/cluster_gateways.go

```diff
@@ -64,45 +59,10 @@ func (c *Cluster) CreateGateway(ctx context.Context, params CreateGatewayParams)
 		Insecure:              c.clusterClient.InsecureSkipVerify,
 		WebProxyAddr:          c.clusterClient.WebProxyAddr,
 		Log:                   c.Log.WithField("gateway", params.TargetURI),
-	})
+	}, cliCommandProvider)
```

Comment:
Same question for this one: should we include the `cliCommandProvider` in the `gateway.Config`?

Reply:
Re: what I said earlier about moving CLI command provider within the struct: in this case the downside of having it in the Config struct is that it'd be public because of how struct embedding works. But I think it makes sense to encapsulate it within the gateway.

For now I'll keep it as a separate arg then, unless you have a suggestion on how this could be approached in a different way.

Reply:
> downside of having it in the Config struct is that it'd be public 

I think that's ok :thinking:
This `gateway.Config` only exists here - It's created and immediately used
Having one extra public field should cause no harm
Why do you say it is a downside?

Reply:
`Gateway` is a long-living object, stored inside the daemon, so it's available across many different requests. `Config` is embedded within `Gateway`, so this would make `CLICommandProvider` public.

From the time I was writing object-oriented code in Ruby and JavaScript, it was a good practice to make your collaborators private, the ones that you receive during object initalization.

I think this idea still makes sense here – why make `CLICommandProvider` visible to anyone using a `Gateway` if `Gateway` only needs it for one specific thing? I think doing that helps to enforce some boundaries in the code.

Other than during initalization, places interacting with `Gateway` shouldn't really now that it uses `CLICommandProvider` underneath. Making it private helps to enforce this constraint.

Reply:
Yeah, it makes sense
Thank you :pray: